blockchain/blockchain.go

```diff
@@ -1763,38 +1756,40 @@ func (bc *BlockChain) CloseBlockSubscriptionLoop() {
 }
 
 // replaceCurrentBlock replaces bc.currentBlock to the given block.
-// func (bc *BlockChain) replaceCurrentBlock(latestBlock *types.Block) {
-// 	bc.mu.Lock()
-// 	defer bc.mu.Unlock()
-//
-// 	if latestBlock == nil {
-// 		logger.Error("no latest block")
-// 		return
-// 	}
-//
-// 	// Return early if it is the first block update.
-// 	currentBlock := bc.CurrentBlock()
-// 	if currentBlock == nil {
-// 		bc.insert(latestBlock)
-// 		logger.Info("inserted an initial block", "blockNumber", latestBlock.NumberU64(),
-// 			"hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
-// 		return
-// 	}
-//
-// 	// Don't update current block if the latest block is not newer than current block.
-// 	if currentBlock.NumberU64() >= latestBlock.NumberU64() {
-// 		logger.Debug("")
-// 		return
-// 	}
-//
-// 	// Insert a new block and update metrics
-// 	bc.insert(latestBlock)
-// 	headBlockNumberGauge.Update(latestBlock.Number().Int64())
-// 	bc.stateCache.TrieDB().UpdateMetricNodes()
-//
-// 	logger.Info("inserted a new block", "prevBlockNumber", currentBlock.NumberU64(), "blockNumber",
-// 		latestBlock.NumberU64(), "hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
-// }
+func (bc *BlockChain) replaceCurrentBlock(latestBlock *types.Block) {
+	bc.mu.Lock()
+	defer bc.mu.Unlock()
+
+	if latestBlock == nil {
+		logger.Error("no latest block")
+		return
+	}
+
+	// Return early if it is the first block update.
+	currentBlock := bc.CurrentBlock()
+	if currentBlock == nil {
+		bc.insert(latestBlock)
+		logger.Info("inserted an initial block", "blockNumber", latestBlock.NumberU64(),
+			"hash", latestBlock.Hash().String(), "stateRoot", latestBlock.Root().String())
+		return
+	}
+
+	// Don't update current block if the latest block is not newer than current block.
+	if currentBlock.NumberU64() >= latestBlock.NumberU64() {
+		logger.Debug("")
```

Comment:
```suggestion
		logger.Warn("can not update past block","current", currentBlock.NumberU64(), "latest", latestBlock.NumberU64()) ~~~~
```

Reply:
Isn't it an error case? Is this something that can happen?

Reply:
I changed this log level to `Warn`. @ehnuje It is for a rare case when the fetcher node rewinds its block.